R/read_fwf.R

```diff
@@ -31,6 +31,11 @@ read_fwf <- function(file, col_positions, col_types = NULL,
                      locale = default_locale(), na = c("", "NA"),
                      skip = 0, n_max = Inf, guess_max = min(n_max, 1000), progress = interactive()) {
   ds <- datasource(file, skip = skip)
+
+  if (inherits(ds, "source_file") && empty_file(ds[[1]])) {
```

Comment:
Yeah, it ends up being the same thing as `ds[[1]]`, but it is probably more clean to use file here, thanks.